src/main/java/uk/gov/hmcts/ccd/data/definition/CachedCaseDefinitionRepository.java

```diff
@@ -49,12 +50,13 @@ public CachedCaseDefinitionRepository(@Qualifier(DefaultCaseDefinitionRepository
     @Override
     public CaseTypeDefinition getCaseType(final String caseTypeId) {
         CaseTypeDefinitionVersion latestVersion = this.getLatestVersion(caseTypeId);
-        return caseDefinitionRepository.getCaseType(latestVersion.getVersion(), caseTypeId);
+        return this.getCaseType(latestVersion.getVersion(), caseTypeId);
     }
 
     @Override
     public CaseTypeDefinition getCaseType(int version, String caseTypeId) {
-        return caseDefinitionRepository.getCaseType(version, caseTypeId);
+        return caseTypes.computeIfAbsent(caseTypeId + version,
```

Comment:
Services sometimes use case types with numeric postfixes.
ct1 version 5 will be colliding with ct version 15.
So, better use a key like caseTypeId+"___"+version

Reply:
Corrected.